File: Daily_Practice/78_Subsets.py
class Solution(object):
    def subsets(self, nums):
        """
        :type nums: List[int]
        :rtype: List[List[int]]
        """
        res = [[]]
        for n in nums:
            for i in range(len(res)):
                res.append(res[i] + [n])
        return res

ls = [1,2,3,4]
obj = Solution().subsets(ls)
print(obj)

File: Daily_Practice/78_Subsets.go
package main

import "fmt"

func subsets(nums []int) [][]int {
	return helper(nums, 0)
}

func helper(nums []int, p int) [][]int {
	var result [][]int
	result = append(result, []int{})
	if p == len(nums) {
		return result
	}

	for i := p; i < len(nums); i++ {
		temp := helper(nums, i+1)
		for j := range temp {
			t := append([]int{nums[i]}, temp[j]...)
			result = append(result, t)
		}
	}

	return result
}
func main() {
	var nums = []int{1, 2, 3, 4}
	obj := subsets(nums)
	fmt.Println(obj)
}
